templates/repo/diff/options_dropdown.tmpl

```diff
@@ -3,7 +3,12 @@
 	<i class="dropdown icon"></i>
 	<div class="menu">
 		<a class="item tiny basic toggle button" data-target="#diff-files">{{.i18n.Tr "repo.diff.show_diff_stats"}}</a>
-		<a class="item" href="{{$.RepoLink}}/pulls/{{.Issue.Index}}.patch" download="{{.Issue.Index}}.patch">{{.i18n.Tr "repo.diff.download_patch"}}</a>
-		<a class="item" href="{{$.RepoLink}}/pulls/{{.Issue.Index}}.diff" download="{{.Issue.Index}}.diff">{{.i18n.Tr "repo.diff.download_diff"}}</a>
+		{{if .Issue.Index}}
+			<a class="item" href="{{$.RepoLink}}/pulls/{{.Issue.Index}}.patch" download="{{.Issue.Index}}.patch">{{.i18n.Tr "repo.diff.download_patch"}}</a>
```

Comment:
Why does this go to the issues path page in the first place and not always to `"Commit.ID.String".patch` ?

Reply:
because you can use this feature at PR diff too

Reply:
and at PR diff the last commit diff is useles

Reply:
Ah, now I understand, thanks.